rotkehlchen/api/rest.py

```diff
@@ -4413,3 +4417,44 @@ def reprocess_skipped_external_events(self) -> Response:
             result=_wrap_in_ok_result({'total': total, 'successfull': successfull}),
             status_code=HTTPStatus.OK,
         )
+
+    def export_history_events(
+            self,
+            filter_query: HistoryBaseEntryFilterQuery,
+            directory_path: Path,
+    ) -> Response:
+        dbevents = DBHistoryEvents(self.rotkehlchen.data.db)
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            history_events, _, _ = dbevents.get_history_events_and_limit_info(
+                cursor=cursor,
+                filter_query=filter_query,
+                has_premium=True,
+                entries_limit=None,
+            )
+
+        if len(history_events) == 0:
+            return api_response(
+                wrap_in_fail_result('No history processed in order to perform an export'),
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+        serialized_history_events = []
+        headers = {}
+        for event in history_events:
+            serialized_event = event.serialize()
+            balance = serialized_event.pop('balance')
+            serialized_event['balance_amount'] = balance['amount']
+            serialized_event['balance_usd_value'] = balance['usd_value']
+            serialized_history_events.append(serialized_event)
+            headers.update(serialized_event)
```

Comment:
I tried with the set first, but it was unordered, I wanted it to maintain a fairly constant arrangement in the csv

Reply:
For that you can use the dict and `headers.update({key: None for x in serialized_event.keys()})` since in python 3.7 and forward (not sure if 3.6) dicts keep the inserion order. But add a comment that is for this reason. Otherwise you are storing extra information that you don't need and is not a reference but the actual values I fear